lib/enumerables.rb

```diff
@@ -0,0 +1,177 @@
+module Enumerable
+  # my_each
+  def my_each(&block)
+    arr = self
+    arr = arr.to_a if self.class == Range || arr.is_a?(Hash)
+    count = 0
+    if arr.is_a?(Hash)
+      while count < arr.length
+        yield arr[i][0], arr[i][1]
+        count += 1
+      end
+      self
+    elsif block
+      while count < arr.length
+        yield arr[count]
+        count += 1
+      end
+      self
+    else
+      "\#<Enumerator: #{self}:my_each>"
+    end
+  end
+
+  # my_each_with_index
+  def my_each_with_index(&block)
+    arr = self
+    arr = arr.to_a if self.class == Range || arr.is_a?(Hash)
+    count = 0
+    if arr.is_a?(Hash)
+      while count < arr.length
+        yield arr[i], count
+        count += 1
+      end
+      self
+    elsif block
+      while count < arr.length
+        yield arr[count], count
+        count += 1
+      end
+      self
+    else
+      "\#<Enumerator: #{self}:my_each_with_index>"
+    end
+  end
+
+  # my_select
+  def my_select(&block)
+    return "\#<Enumerator: #{self}:my_select>" unless block
+
+    temp = []
+    to_a.my_each { |num| temp.push(num) if yield(num) }
+    temp
+  end
+
+  # my_all
+  def my_all?(paramet = nil)
+    arr = self
+    arr = arr.to_a
+    return true if arr.empty?
+
+    if block_given?
+      arr.my_each do |n|
+        ans = yield n
+        return false if ans == false
+      end
+    elsif paramet.nil?
+      arr.my_each { |n| return false if n.nil? || !n }
+    elsif paramet.is_a?(Regexp)
+      arr.my_each { |x| return false unless x.include?(paramet) }
+    elsif paramet.is_a?(Class)
+      arr.my_each { |x| return false unless x.is_a?(paramet) }
+    else
+      arr.my_each { |x| return false if x != paramet }
+    end
+    true
+  end
+
+  # my_any
+  def my_any?(paramet = nil)
+    arr = self
+    arr = arr.to_a
+    return true if arr.empty?
+
+    if block_given?
+      arr.my_each do |n|
+        ans = yield n
+        return true if ans == true
+      end
+    elsif paramet.nil?
+      arr.my_each { |n| return true if !!n == true && !n.nil? }
+    elsif paramet.is_a?(Regexp)
+      arr.my_each { |x| return true if x.match(paramet) }
+    elsif paramet.is_a?(Class)
+      arr.my_each { |x| return true if x.include?(paramet) }
+    else
+      arr.my_each { |x| return true if x == paramet }
+    end
+    false
+  end
+
+  # my_none
+  def my_none?(paramet = nil)
+    arr = self
+    arr = arr.to_a
+    return true if arr.empty?
+
+    if block_given?
+      arr.my_each do |n|
+        ans = yield n
+        return false if ans == true
+      end
+    elsif paramet.nil?
+      arr.my_each { |n| return false if !!n == true && !n.nil? }
+    elsif paramet.is_a?(Regexp)
+      arr.my_each { |x| return false if x.match(paramet) }
+    elsif paramet.is_a?(Class)
+      arr.my_each { |x| return false if x.include?(paramet) }
```

Comment:
Same problem as before: `#my_each` isn't prepared for Class values:

```
  8) enumerables #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
     Failure/Error: arr.my_each { |x| return false if x.include?(paramet) }
     
     NoMethodError:
       undefined method `include?' for 2:Integer
     # ./lib/enumerables.rb:117:in `block in my_none?'
     # ./lib/enumerables.rb:15:in `my_each'
     # ./lib/enumerables.rb:117:in `my_none?'
     # ./spec/enumerables_spec.rb:306:in `block (4 levels) in <top (required)>'

  9) enumerables #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric
     Failure/Error: arr.my_each { |x| return false if x.include?(paramet) }
     
     NoMethodError:
       undefined method `include?' for 1:Integer
     # ./lib/enumerables.rb:117:in `block in my_none?'
     # ./lib/enumerables.rb:15:in `my_each'
     # ./lib/enumerables.rb:117:in `my_none?'
     # ./spec/enumerables_spec.rb:310:in `block (4 levels) in <top (required)>'

  10) enumerables #my_none? when a class is passed as an argument returns false if any of the collection is a member of such class
      Failure/Error: arr.my_each { |x| return false if x.include?(paramet) }
      
      TypeError:
        no implicit conversion of Class into String
      # ./lib/enumerables.rb:117:in `include?'
      # ./lib/enumerables.rb:117:in `block in my_none?'
      # ./lib/enumerables.rb:15:in `my_each'
      # ./lib/enumerables.rb:117:in `my_none?'
      # ./spec/enumerables_spec.rb:315:in `block (4 levels) in <top (required)>'
```

Please fix this in the same way as before.